resource_obmcs_core_volume.go

```diff
@@ -196,9 +200,22 @@ func (s *VolumeResourceCrud) Create() (e error) {
 
 func (s *VolumeResourceCrud) Get() (e error) {
 	res, e := s.Client.GetVolume(s.D.Id())
-	if e == nil {
-		s.Res = res
+
+	// IsHydrated will be false (null) for regular block volumes in the provisioning phase,
+	// don't set state and continue polling until provisioning->available, when IsHydrated
+	// also becomes true.
+	//
+	// IsHydrated will be false for clones in the provisioning phase, don't set state and continue polling
+	// until provisioning->available, then:
+	// IsHydrated may still be false for clones after becoming available, still dont set state and
+	// continue polling until IsHydrated is true. This prevents 409 errors when trying to delete a volume
+	// while it is still hydrating.
+	if e != nil || !res.IsHydrated {
```

Comment:
This makes me a little bit nervous - according to the docs, https://docs.us-phoenix-1.oraclecloud.com/api/#/en/iaas/20160918/Volume/, isHydrated "Specifies whether the cloned volume's data has finished copying from the source volume or backup." That makes it sound like maybe it wouldn't be returned if not using cloning. Even if the behavior we're seeing is that it is set for all volumes, I would want the service team to confirm that. Or is there some way we can tell whether it's a clone?

Reply:
I'm a bit nervous for a different reason.  This change is adding extra semantics to "Get()".  I'm not sure it's okay for general use cases to have this implication.  Is it guaranteed by terraform that "Get()" is always called before "Delete()"?